### ISSUE TITLE
Feature: Text and URL Drag-and-Drop

### ISSUE THREAD

**Issue by Joybuke** [alacritty/alacritty#4914]
Right now, the only content that can be dropped into the terminal is files. It would be nice if text and URLs are able to be dropped in, similar to how how mate-terminal works. Though after trying to implement this myself, (even though I'm a total Rust noob) it seems that `WindowEvent::DroppedFile` is not able to accept text or URLs in the way that this is desired to easily allow this.

**Comment by chrisduerr** [comment]
I don't see any benefit to implementing this, especially with the existence of the selection clipboard on Linux.

If you want to implement this yourself, you'd probably need to do so in https://github.com/rust-windowing/winit though. If you're interested in this feel free to open an issue there.

**Comment by Joybuke** [comment]
I would say there's benefit for those working with GUI applications that don't easily allow copying and pasting from other sources (such as some less-than-friendly gui applications).  Though due to how the library is structured, I can understand avoidance of this. If I can get some time and some help from the winit devs, I'll surely make a PR for it. Thanks for being diligent with a response though!

**Comment by chrisduerr** [comment]
I honestly don't see how dragging a text that likely must be selected first anyways would be any more convenient than using the selection clipboard.

**Comment by Joybuke** [comment]
Well to give an example from my work flow, my work uses Discord for communication on projects, specifically art and music work. If I need to grab multiple assets at once, I used to just open a terminal, go to a project folder, and click and drag a bunch of assets into wget. I can't do this as easily now (especially for audio) since I cannot click and drag the download button to get the url for my terminal. There's plenty of other examples of issues too, such as clicking and dragging images from sites that have disabled the right-click function, but the discord example is the most relevant and pressing for me. 

In other news, I found that this issue is being worked by rust-windowing/winit#720.
